### PR TITLE
Improved Nav #25 - Ben Lambert

### PR DESCRIPTION
**Pull request by benlambert43**
- Lifted state up from the NavBar component into the App component. 
 - Added an event handler in the App component for Page changes and passed it down to Navbar
 - Created a Router function in App that conditionally renders different pages based on state

### CODE REVIEW

File: challenge/src/App.jsx [hunk 1]
```diff
@@ -1,45 +1,270 @@
...
+import Fade from "react-reveal/Fade";
 
 class App extends Component {
-    state = {}
-    constructor(props) {
-        super(props);
-    }
-    render() {
-        console.log("Render Method is Working. Redered successfully! I think?")
+  state = {};
```
Comment (mjrb) [inline challenge/src/App.jsx:9]: just a note, please avoid reformatting or reindenting entire files in a pr for some issue, it makes it hard to see where actual changes have been made. instead they should be submitted in a different pr. (and a realy project should be set up with a linter to vet code, but this isn't a real project XD)

File: challenge/src/App.jsx [hunk 2]
```diff
@@ -1,45 +1,270 @@
...
-              <BabyYoda />
-            </>
+          <div>
+            <h1 style={{ fontSize: 100, color: "#FFC436" }}>
+              <center>Sponsors Page</center>
+            </h1>
+            <div style={{ backgroundColor: "whitesmoke" }}>
+              <br />
+            </div>
+            <div
```
Comment (mjrb) [inline challenge/src/App.jsx]: it may also be nice to make a random image component. the random image idea was a nice touch tho

File: challenge/src/App.jsx [hunk 3]
```diff
@@ -1,45 +1,270 @@
...
+  handlePageChange(index) {
+    this.setState({ currentPage: index });
+  }
+
+  Router(param) {
+    console.log(param.PageRoute);
+    switch (param.PageRoute) {
+      case 0:
+        return (
+          <div>
```
Comment (mjrb) [inline challenge/src/App.jsx]: breaking each page into its own component like i think you said in your email would be cool

File: challenge/src/App.jsx [hunk 4]
```diff
@@ -1,45 +1,270 @@
...
+
+  render() {
+    console.log("User has selected page # ", this.state.currentPage);
+    //console.log("Render Method is Working. Redered successfully! I think?")
+
+    return (
+      <>
+        <Navbar
+          fixed={true}
+          items={["Home", "Sponsors", "About", "Contact us"]}
```
Comment (mjrb) [inline challenge/src/App.jsx:52]: the names are passed into the navbar, perhaps instead of the callback giving the index, have the callback give the page name, for a more friendly interface

**Review by mjrb — CHANGES_REQUESTED** [review]
over all pretty good, only a few tweaks I'd like to see. you don't necessarily have to undo the code formatting thing that was just a note.

File: challenge/src/App.jsx (hunk 1 above)
Comment (benlambert43) [reply challenge/src/App.jsx:9]: Sorry about that! I forgot to disable my code formatter before beginning, I will do so from now on.

File: challenge/src/App.jsx (hunk 2 above)
Comment (benlambert43) [reply challenge/src/App.jsx]: I created a random image component in src/components/RandomImage

File: challenge/src/App.jsx (hunk 3 above)
Comment (benlambert43) [reply challenge/src/App.jsx]: I went ahead and did this, the pages are now in src/components/Pages. There is a lot of reused code between each page but in a real website it would make more sense to keep each page separate

File: challenge/src/App.jsx (hunk 4 above)
Comment (benlambert43) [reply challenge/src/App.jsx:52]: I added some state to App.jsx to keep track of the currentPageName as well as the currentPageIndex. Now the console returns the name of the page that the user is on, not the index of the page.

File: challenge/src/App.jsx [hunk 5]
```diff
@@ -1,45 +1,71 @@
...
+    this.handlePageChange = this.handlePageChange.bind(this);
+  }
+
+  handlePageChange(index, indexName) {
+    this.setState({ currentPage: index, currentPageName: indexName });
+  }
+
+  Router(param) {
+    switch (param.PageRoute) {
+      case 0:
```
Comment (mjrb) [inline challenge/src/App.jsx]: my point of changing the onPageChange to pass the name of the page was so we could have `case 'home'` here, its sort of a nit pick at this point though so I'll tweak that really quick and we can move on.

**Review by mjrb — APPROVED** [review]
lgtm
